### ISSUE TITLE
ability to run on aws

### ISSUE THREAD

**Issue by pyramation** [df7cb/postgresql-unit#28]
I absolutely love this plugin. I've been reading through the repo and it's truly amazing.

I believe since this is a C plugin, it's not possible to run on AWS RDS Postgres. I'm interested in potentially converting this to raw domains, types, and stored procedures... I have two questions: 1. is that a bad idea? 2. and if not, where should I start? 

Happy to talk offline but figured I'd open and issue to start the convo.

**Comment by df7cb** [comment]
Hi Dan,

there's several bits that you would need to replace:
* the datatype is currently 16 bytes for (8-array of char(1) for the base unit exponents, double precision value). Replacing that by a standard PostgreSQL compound type would probably work well (the storage format would change, though)
* the logic to parse prefix-unit words like millimeter as milli-meter are already implemented as SPI SQL calls
* but on top of that parser, there is a whole bison grammar for parsing expressions like `9.81 kg/m*s^2`, reimplementing that in pure plpgsql won't be feasible.

So in short, I don't think that will work. (Please reopen if you see some way I didn't see.)
